store/searchlayer/post_layer.go

```diff
@@ -78,9 +78,9 @@ func (s SearchPostStore) Delete(postId string, date int64, deletedByID string) *
 	return err
 }
 
-func (s SearchPostStore) searchPostsInTeamForUserByEngine(engine searchengine.SearchEngineInterface, paramsList []*model.SearchParams, userId, teamId string, isOrSearch, includeDeletedChannels bool, page, perPage int) (*model.PostSearchResults, *model.AppError) {
+func (s SearchPostStore) searchPostsInTeamForUserByEngine(engine searchengine.SearchEngineInterface, paramsList []*model.SearchParams, userId, teamId string, page, perPage int) (*model.PostSearchResults, *model.AppError) {
 	// We only allow the user to search in channels they are a member of.
-	userChannels, err := s.rootStore.Channel().GetChannels(teamId, userId, includeDeletedChannels)
+	userChannels, err := s.rootStore.Channel().GetChannels(teamId, userId, paramsList[0].IncludeDeletedChannels)
```

Comment:
This is fine by now by we should in the future to a) create a new struct called SearchParamsList with the `IncludeDeletedChannels` as member of that struct instead of having it inside the `SearchParams` one

Reply:
I thought introducing something similar to `SearchParamsList` but I thought it may not be ok to introduce it. I can add the struct and make necessary changes

Reply:
@ethervoid I activated tests and they are passing now. Should I work on `SearchParamsList`?

Reply:
No, that should go in other PR because that change involves:

- a) a discussion about it because I'm not 100% sure we should do it and 
- b) a lot of changes in different parts of the code.

Taking into account that we're assuming all the `IncludeDeleteChannels` flags have the same value, we should include some kind of assertion [here](https://github.com/mattermost/mattermost-server/pull/14573/files#diff-c72bb0286efe7214238420b48bc0f77dR83) that checks that all tine include delete are the same // @mgdelacroix

Reply:
@ethervoid sorry, I could not get what you mean. Do you want me to add the assertion right away or you just asked his opinion on whether the assertion should be added?

Reply:
Yeah sorry for the confusion. I was proposing that idea and including Miguel in the conversation to ask for his opinion.

Taking for granted that the `IncludeDeletedChannels` property is the same for all the elements in the slice is a bit risky. So yes, I'd add the assertion until we can perform the refactoring.

Reply:
I agree with the idea of the assertion to ensure that the condition is met before we go for the refactor

Reply:
@mgdelacroix I added the assertion 👍